src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/FieldBuilderImpl.cs

```diff
@@ -9,23 +12,64 @@ internal sealed class FieldBuilderImpl : FieldBuilder
     {
         private readonly TypeBuilderImpl _typeBuilder;
         private readonly string _fieldName;
-        private readonly FieldAttributes _attributes;
+        internal int _offset;
+        private FieldAttributes _attributes;
         private readonly Type _fieldType;
 
+        internal List<CustomAttributeWrapper>? _customAttributes;
+
         internal FieldBuilderImpl(TypeBuilderImpl typeBuilder, string fieldName, Type type, FieldAttributes attributes)
         {
             _fieldName = fieldName;
             _typeBuilder = typeBuilder;
             _fieldType = type;
             _attributes = attributes & ~FieldAttributes.ReservedMask;
+            _offset = -1;
         }
 
-        #region MemberInfo Overrides
         protected override void SetConstantCore(object? defaultValue) => throw new NotImplementedException();
-        protected override void SetCustomAttributeCore(ConstructorInfo con, byte[] binaryAttribute) => throw new NotImplementedException();
+        protected override void SetCustomAttributeCore(ConstructorInfo con, ReadOnlySpan<byte> binaryAttribute)
+        {
+            if (!IsPseudoAttribute(con.ReflectedType!.FullName!, binaryAttribute))
```

Comment:
The method is called `IsPseudoCustomAttribute` but it does more than just checking whether the attribute is a pseudo custom attribute. `HandlePseudoCustomAttribute` or similar name may be more appropriate. 

Alternatively, handling of the pseudo custom attributes does not really need to be a separate method - something like this would be fine too:

void SetCustomAttributeCore(ConstructorInfo con, ReadOnlySpan<byte> binaryAttribute)
{
    // Handle pseudo custom attributes
    switch (con.ReflectedType!.FullName!)
    {
    case XXX:
        ...
        return;

    }

    _customAttributes ??= new List<CustomAttributeWrapper>();
    _customAttributes.Add(new CustomAttributeWrapper(con, binaryAttribute));
}
```